src/utils/template_handler.rs

```diff
@@ -129,44 +159,73 @@ pub(crate) fn load_remote_template_collection(path: &str, url: &str, force: bool
                     .replace('"', "")
                     .as_str(),
                 force,
+                &url_type,
             );
             if !st.is_ok {
                 return st;
             }
         }
     }
+
     Status::ok()
 }
 
-/// Load a template from a remote repository
-pub(crate) fn load_remote_template(path: &str, url: &str, force: bool) -> Status {
-    if !force && Path::new(path).exists() {
-        return Status::error(format!(
-            "Template {} already exists...",
-            path.replace(".templates/", "")
-        ));
-    }
+/// Load a template from a gitlab repository
+pub(crate) fn load_gitlab_template(
+    response: serde_json::Value,
+    path: &str,
+    url: &str,
+    force: bool,
+) -> Status {
+    let items = response.as_array().unwrap();
 
-    if !Path::new(path).exists() {
-        std::fs::create_dir(path).unwrap();
-    }
+    for item in items {
+        if item["type"] == "tree" {
+            let st = load_remote_gitlab_template_dir(
+                format!("{}/{}", path, item["name"])
```

Comment:
Scince this formatting is done here twice maybe add a lambda just before and then u just have to apply the lambda to the path and the url.

Reply:
Sure @cophilot , Instead of Lambda function, I extracted a whole new function for this fomatting purpose, and put those at the last of `template_handler.rs`  only. As it is using in many other places other than this, as pointed by you.